substratevm/src/com.oracle.svm.jfr/src/com/oracle/svm/jfr/traceid/JfrTraceId.java

```diff
@@ -0,0 +1,140 @@
+/*
+ * Copyright (c) 2020, 2021, Oracle and/or its affiliates. All rights reserved.
+ * Copyright (c) 2020, 2021, Red Hat Inc. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package com.oracle.svm.jfr.traceid;
+
+import org.graalvm.nativeimage.Platform;
+import org.graalvm.nativeimage.Platforms;
+
+import com.oracle.svm.core.annotate.Uninterruptible;
+
+import jdk.jfr.internal.Type;
+
+/**
+ * When a class is referenced in an event, the unique ID of that class is tagged as in-use for the
+ * current JFR epoch.
+ */
+public class JfrTraceId {
+
+    public static final long BIT = 1;
+    public static final long META_SHIFT = 8;
+    public static final long TRANSIENT_META_BIT = (BIT << 3);
+    public static final long TRANSIENT_BIT = (TRANSIENT_META_BIT << META_SHIFT);
+    public static final long SERIALIZED_META_BIT = (BIT << 4);
+    public static final long SERIALIZED_BIT = (SERIALIZED_META_BIT << META_SHIFT);
```

Comment:
It doesn't seem to me like `TRANSIENT_BIT` or `SERIALIZED_BIT` can ever be set within a tag.

If so, then these could be deleted, as well as the checks for the `TRANSIENT_BIT`

Reply:
Right, those are unused in this incarnation. We might need them later, when (and if) we implement stuff like leak detection, but should probably add them back only when we actually need them. They can be safely removed.